reactive/docker_registry.py

```diff
@@ -241,15 +253,19 @@ def update_reverseproxy_config():
    - balance leastconn
    - option httpchk GET / HTTP/1.0
   servers:
-   - [%(unit)s, %(addr)s, %(port)s, 'check port %(port)s']
+   - [%(unit)s, %(addr)s, %(port)s, %(server_opts)s]
```

Comment:
When I ran my tests yesterday, the haproxy units appeared to be completely ignoring the server options configured via the relation. If you have this working, I'd be curious to see your haproxy charm config.

Reply:
@axinojolais i just used the default charm config you get with `juju deploy haproxy`, most notably it's in `active-passive` mode. I think the key change that made these server opts valid in this PR was the following from a few lines below:
```diff
- website.set_remote(all_services=services_yaml)
+ website.set_remote(services=services_yaml)
```

According to [charm docs](https://bazaar.launchpad.net/~haproxy-team/charm-haproxy/trunk/view/head:/README.md#L108), `all_services` is meant to be used when haproxy itself is related to a rev proxy like apache2. We were passing this key from the registry to **haproxy's** revproxy relation, and it was smashing all the service entries under things like this:

```
frontend haproxy-0-80
    bind 0.0.0.0:80
    default_backend haproxy_service

backend haproxy_service
    balance leastconn
    cookie SRVNAME insert
    server drold-0-5000 172.31.24.31:5000 maxconn 100 cookie S0 check
```
Note the service name there is `haproxy-x-80` versus `docker-registry-x`.  This works by accident when you have 1 haproxy, but when you add another unit, you get a config like this:
```
frontend haproxy-0-81
    bind 0.0.0.0:81
    default_backend haproxy_service_be

backend haproxy_service_be
    balance leastconn
    cookie SRVNAME insert
    server drold-0-5000 172.31.24.31:5000 maxconn 100 cookie S0 check

frontend haproxy-0-80
    bind 0.0.0.0:80
    default_backend haproxy_service
    mode tcp
    option tcplog

backend haproxy_service
    mode tcp
    option tcplog
    balance leastconn
    server haproxy-0 172.31.15.127:81 check
    server haproxy-1 172.31.28.222:81 check backup
```

Now we have a mix of the registry services as well as haproxy's peer services, all smashed around under some form of the `haproxy-x` service name.  By sending `services` instead of `all_services`, this PR gives us an haproxy config like this:
```
frontend haproxy-0-5001
    bind 0.0.0.0:5001
    default_backend registry_be
    mode http

backend registry_be
    mode http
    balance leastconn
    option httpchk GET / HTTP/1.0
    server registry-0 172.31.4.3:5000 check inter 2000 rise 2 fall 5 maxconn 4096
    server registry-1 172.31.16.65:5000 check inter 2000 rise 2 fall 5 maxconn 4096 backup

frontend haproxy-0-5000
    bind 0.0.0.0:5000
    default_backend registry
    mode tcp
    option tcplog

backend registry
    mode tcp
    option tcplog
    balance leastconn
    server haproxy-0 172.31.11.140:5001 check
    server haproxy-1 172.31.31.180:5001 check backup

frontend haproxy-0-81
    bind 0.0.0.0:81
    default_backend haproxy_service_be

backend haproxy_service_be
    balance leastconn
    cookie SRVNAME insert

frontend haproxy-0-80
    bind 0.0.0.0:80
    default_backend haproxy_service
    mode tcp
    option tcplog

backend haproxy_service
    mode tcp
    option tcplog
    balance leastconn
    server haproxy-0 172.31.11.140:81 check
    server haproxy-1 172.31.31.180:81 check backup
```
This feels right to me, but I'll admit that haproxy is a black box to me most days. At least the `service_name` and `server_opts` are making it into the config.